File: defect_report_mobile/lib/Models/register_dto.dart
class RegisterDto {
  final String name;
  final String email;
  final String password;
  final String role;

  RegisterDto({
    required this.name,
    required this.email,
    required this.password,
    required this.role,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'email': email,
        'password': password,
        'role': role,
      };
}

File: DefectApi/Dto/RegisterDto.cs
using System;
using System.ComponentModel.DataAnnotations;
namespace DefectApi.Dto
{
    public class RegisterDto
    {
        public string Name { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        [Required]
        [Compare("Password", ErrorMessage = "Passwords do not match.")]
        public string ConfirmPassword { get; set; }

        public string Role { get; set; }
    }
}